dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java

```diff
@@ -230,6 +232,19 @@ private static void generateWitnessIfAble(VerificationTask task, ProverEnvironme
         }
     }
 
+    private static String getLtlPropertyFromSummary(String summary) {
+        if(summary.contains("integer overflow")) {
+            return "CHECK( init(main()), LTL(G ! overflow))";
+        }
+        if(summary.contains("invalid dereference")) {
+            return "CHECK( init(main()), LTL(G valid-deref))";
+        }
+        if(summary.contains("user assertion")) {
+            return "CHECK( init(main()), LTL(G ! call(reach_error())))";
+        }
+        throw new UnsupportedOperationException("Violation found for unsupported property");
+    }
```

Comment:
I think this method makes more sense in the `WitnessBuilder` since it is totally SVCOMP-specific.
The `WitnessBuilder.of` factory method can extract the LTL formula from the summary.